### PR TITLE
ci: Test minimum supported BTC version

### PR DESCRIPTION
**Pull request by ShahanaFarooqui**
As suggested by @rustyrussell, Adding another step for testing minimum supported Bitcoin version with clang test suite.

Changelog-None.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -446,6 +446,77 @@ jobs:
         run: |
           poetry run pytest tests/ -vvv -n 2 ${PYTEST_OPTS}  ${{ matrix.PYTEST_OPTS }}
 
+  min-btc-support:
+    name: Test BTC version 25.0 ${{ matrix.name }}
```
Comment (fanquake) [inline .github/workflows/ci.yaml:450]: Could test `25.2` here instead? That is the latest release on the `25.x` branch.

Also, looking at https://github.com/ElementsProject/lightning?tab=readme-ov-file#getting-started, I guess the minimum required verison is now 25.x rather than 22.0?

Comment (ShahanaFarooqui) [reply .github/workflows/ci.yaml:450]: @fanquake Thanks for your review. I've updated the README accordingly. However, maintaining the minimum supported BTC version at 25.0 to accommodate versions released till two years ago. Version 25.0 was released in May 2023, while version 25.2 was released in April 2024.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -446,6 +446,77 @@ jobs:
...
+      - compile
+    strategy:
+      fail-fast: true
+      matrix:
+        include:
+          - NAME: clang
+            CFG: compile-clang
+            TEST_DB_PROVIDER: sqlite3
+            COMPILER: clang
+            TEST_NETWORK: regtest
```
Comment (niftynei) [inline .github/workflows/ci.yaml:466]: nit: put the BITCOIN_VERSION as a matrix value instead of repeatedly putting it into the definition below.

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -446,6 +446,77 @@ jobs:
...
+        with:
+          python-version: 3.8
+
+      - name: Install dependencies
+        run: |
+          pip3 install --user pip wheel poetry
+          poetry install
+
+      - name: Download Bitcoin Core
+        run: wget "https://bitcoincore.org/bin/bitcoin-core-25.0/bitcoin-25.0-x86_64-linux-gnu.tar.gz"
```
Comment (niftynei) [inline .github/workflows/ci.yaml:482]: this should be 

        run: wget "https://bitcoincore.org/bin/bitcoin-core-${{ matrix.BITCOIN_VERSION }}/bitcoin-${{ matrix.BITCOIN_VERSION }}-x86_64-linux-gnu.tar.gz"

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -446,6 +446,77 @@ jobs:
...
+
+      - name: Extract Bitcoin Core
+        run: tar -xf "bitcoin-25.0-x86_64-linux-gnu.tar.gz"
```
Comment (niftynei) [inline .github/workflows/ci.yaml:485]: same as above

File: .github/workflows/ci.yaml [hunk 5]
```diff
@@ -446,6 +446,77 @@ jobs:
...
+
+      - name: Move Bitcoin Core Binaries
+        run: sudo mv bitcoin-25.0/bin/* /usr/local/bin/
```
Comment (niftynei) [inline .github/workflows/ci.yaml:488]: same as above

File: .github/workflows/ci.yaml [hunk 6]
```diff
@@ -446,6 +446,77 @@ jobs:
...
+
+      - name: Clean Up Downloaded Bitcoin
+        run: rm -rf "bitcoin-25.0-x86_64-linux-gnu.tar.gz" "bitcoin-25.0"
```
Comment (niftynei) [inline .github/workflows/ci.yaml:491]: same as above

**Review by niftynei — COMMENTED** [review]
Parameterize the version number for the job?

**Comment by niftynei** [comment]
documentation on matrix values for github CI jobs: https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (niftynei) [inline .github/workflows/ci.yaml:450]: note that the `matrix.name` here is "clang" (see L462). Not sure if that's very useful info to include in the test name?

would be more useful to change the `25.0` to use the parameterized bitcoin version instead?

for example

```
name: Test min-bitcoind v${{ matrix.bitcoin_version }}
```

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (ShahanaFarooqui) [reply .github/workflows/ci.yaml:466]: Fixed by adding MIN_BTC_VERSION for the step.

File: .github/workflows/ci.yaml (hunk 4 above)
Comment (ShahanaFarooqui) [reply .github/workflows/ci.yaml]: Fixed with env MIN_BTC_VERSION.

File: .github/workflows/ci.yaml (hunk 5 above)
Comment (ShahanaFarooqui) [reply .github/workflows/ci.yaml]: Fixed with env MIN_BTC_VERSION.

File: .github/workflows/ci.yaml (hunk 6 above)
Comment (ShahanaFarooqui) [reply .github/workflows/ci.yaml]: Fixed with env MIN_BTC_VERSION.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (ShahanaFarooqui) [reply .github/workflows/ci.yaml]: I added the matrix.name in the title to clearly show that the step has been tested with `clang` only. But I have updated the title to be more clearer with `Test minimum supported BTC v${{ matrix.MIN_BTC_VERSION }} with ${{ matrix.NAME }}`.

**Review by niftynei — APPROVED** [review]
ACK 5a428b00a1c8d15ad28c6a1d82e80b88a1b45899
